src/meta/src/hummock/manager/commit_epoch.rs

```diff
@@ -486,4 +300,79 @@ impl HummockManager {
             }
         }
     }
+
+    async fn correct_commit_ssts(
+        &self,
+        sstables: Vec<LocalSstableInfo>,
+        table_compaction_group_mapping: &HashMap<TableId, CompactionGroupId>,
+    ) -> Result<BTreeMap<u64, Vec<SstableInfo>>> {
+        let mut new_sst_id_number = 0;
+        let mut sst_to_cg_vec = Vec::with_capacity(sstables.len());
+        for commit_sst in sstables {
```

Comment:
In this for loop, we can collect these per group information
```
HashMap<
   CompactionGroupId,
   (
        Vec<(
          SstableInfo, 
          u64 /* sstable_id_offset from new_sst_id,
so that we don't have to increment new_sst_id, 
but instead get the sstable id by new_sst_id + sstable_id_offset */
        )>, 
        HashSet<TableId>
    )
>
```

And then in the next round of processing, we can handle the ssts of a compaction group all at once.

Reply:
I tried refactoring the code with this complex structure, but the code was not simplified.

Since `next_sstable_object_id` does not implement prefetching under sql backend, this may become burdensome when we can't estimate the total sst_number (which can't be done using only one loop), so I'm biased towards leaving this code untouched for now, what do you think?